modules/core/src/main/scala/com.snowplowanalytics.iglu/client/resolver/ResolverCache.scala

```diff
@@ -118,21 +156,31 @@ object ResolverCache {
       Applicative[F].pure(none)
   }
 
-  def getItem[F[_]: Monad: Clock, A, K](
+  private[resolver] def shouldCreateResolverCache(size: Int, ttl: Option[TTL]): Boolean =
```

Comment:
On the 1.x branch the criteria seems to be size > 0 and if there is a TTL, then it should not be a negative number
On the 2.x branch, the criteria seems to be size >= 0 and if there is a TTL, then it should not be a negative number:
https://github.com/snowplow/iglu-scala-client/commit/d1e1a37fc05f1710ef75fd893279a8578bc8435b#diff-9e946c350a6fbf56aa3e5fe99455e9cd3d3d2dd2296d5a4a04f03bf6a060eaf2R112

but I don't think it should be possible to create a cache with cache size 0 so this change brings back the 1.x criteria

see: https://github.com/snowplow/iglu-scala-client/pull/211/files#diff-c074249fe61ae962f11d3dfc7a272de9de483bf50b281cdfd2524da3800899f1R105-R111